stdlib/src/array.rs

```diff
@@ -1281,10 +1282,22 @@ mod array {
     pub struct PyArrayIter {
         position: AtomicUsize,
         array: PyArrayRef,
+        internal: PyMutex<PositionIterInternal<PyArrayRef>>,
```

Comment:
do you really need both `position` and `internal`? Though I didn't look in details, those fields look like duplication.

Reply:
The PositionIterInternal was used by referring to the implementation of the list iter.

Reply:
Is it because `internal` contains position because it is `PositionIterInternal` as its name suggests?

Reply:
Check the `PositionIterInternal` type. it contains `position` field. By this change, the code have duplicated position fields but most of code uses position and new code uses internal. That can't work correctly.

Reply:
With comparison with other types' iterator, we really don't need position and array field because `PyMutex<PositionIterInternal<PyArrayRef>>` will do their roles. Because of `__next__` and `__setstate__`, we need to modify `position` and `array` fields. It leads to the need for `PyMutex`.

Reply:
If we replace `position` and `array` fields with `PyMutex<PositionIterInternal<PyArrayRef>>`, we can implement other methods as usual we did (`__next__` and `__reduce__`)

Reply:
Does arrayiterator require `closed` field? I thought it only need `position`.
`closed` is useful when it include another iterator in it.